### ISSUE TITLE
plan and job need spec.display_name and need to support sorting and filtering it

### ISSUE THREAD

**Issue by fuchunlan** [jibutech/backup-saas-client#77]
When a  is deleted, sometimes its jobs are not deleted, if create a plan with the same name，the previous job will be added to the list, and PV will be replaced,

Perhaps we could set metadata.name to uuid and the page display name to display_name

**Comment by fuchunlan** [comment]
@shaofeng66  maybe we don't need to add display_name, I found there is `metadata.uid`,could we replace the job query : plan name with the `metadata.uid` ?
This is my idea, not sure if it is feasible, can you evaluate it？
![image](https://user-images.githubusercontent.com/63338728/145946787-65d2b912-d3bb-4b9e-8cd7-056fc8370713.png)

**Comment by shaofeng66** [comment]
@fuchunlan The metadata.uid is a field on backupplan, not a field on job like job.Spec.BackupName, it would need n extra querys to get the uids and do the filter then sorting for n jobs. There would be big performance impact to rest server and api server.
Setting uuid on name will also have a side effect that the status.jobDetail.LatestJobName will be an uuid, I see in current backupplan page, this name is not shown, so I think it might be OK for now.
I think we should ensure jobs are all deleted when plan is deleted, would it help if we support cascading deletion in  issue #15 ?

**Comment by fuchunlan** [comment]
> 

Yes, if we can ensure jobs are all deleted when plan is deleted, this issue is unnecessary.

**Comment by fuchunlan** [comment]
> @fuchunlan The metadata.uid is a field on backupplan, not a field on job like job.Spec.BackupName, it would need n extra querys to get the uids and do the filter then sorting for n jobs. There would be big performance impact to rest server and api server. Setting uuid on name will also have a side effect that the status.jobDetail.LatestJobName will be an uuid, I see in current backupplan page, this name is not shown, so I think it might be OK for now. I think we should ensure jobs are all deleted when plan is deleted, would it help if we support cascading deletion in issue #15 ?

Yes, if we can ensure jobs are all deleted when plan is deleted, this issue is unnecessary.
